### PR TITLE
Fix/aem resource aem id handling

### PR DESCRIPTION
**Pull request by mbloch1986**
* Improved aem_id handling in action_manifests
* Updated path for package download in action_download manifest

### CODE REVIEW

**Comment by mbloch1986** [comment]
PR regarding issue https://github.com/shinesolutions/puppet-aem-resources/issues/27

File: manifests/action_deploy_artifact.pp [hunk 1]
```diff
@@ -13,13 +14,20 @@
   $path              = '/tmp/shinesolutions/aem-aws-stack-provisioner',
 ) {
 
+  $_aem_id = $aem_id ? {
```
Comment (cliffano) [inline manifests/action_deploy_artifact.pp]: Please replace with `pick` function https://github.com/puppetlabs/puppetlabs-stdlib#pick .

File: manifests/action_download_artifacts.pp [hunk 2]
```diff
@@ -118,15 +121,26 @@
   }
 
   $packages.each | Integer $index, Hash $package| {
+
+    if $package[aem_id] {
```
Comment (cliffano) [inline manifests/action_download_artifacts.pp]: Please replace with `pick` function https://github.com/puppetlabs/puppetlabs-stdlib#pick .

File: manifests/action_export_backup.pp [hunk 3]
```diff
@@ -13,22 +14,29 @@
   $data_bucket_name = $::data_bucket_name,
 ) {
 
-  file { "${tmp_dir}/${package_group}":
+  $_aem_id = $aem_id ? {
```
Comment (cliffano) [inline manifests/action_export_backup.pp]: Please replace with `pick` function https://github.com/puppetlabs/puppetlabs-stdlib#pick .

File: manifests/action_export_backups.pp [hunk 4]
```diff
@@ -63,21 +65,32 @@
...
+  $aem_id,
   $backup_path,
   $packages,
   $package_version,
 ) {
 
   $packages.each | Integer $index, Hash $package| {
 
-    if !defined(File["${tmp_dir}/${package[group]}"]) {
+    if $package[aem_id] {
```
Comment (cliffano) [inline manifests/action_export_backups.pp]: Please replace with `pick` function https://github.com/puppetlabs/puppetlabs-stdlib#pick .

File: manifests/action_import_backup.pp [hunk 5]
```diff
@@ -12,11 +13,18 @@
   $data_bucket_name    = $::data_bucket_name,
 ) {
 
+  $_aem_id = $aem_id ? {
```
Comment (cliffano) [inline manifests/action_import_backup.pp]: Please replace with `pick` function https://github.com/puppetlabs/puppetlabs-stdlib#pick .

File: manifests/action_deploy_artifact.pp (hunk 1 above)
Comment (mbloch1986) [reply manifests/action_deploy_artifact.pp]: Done.

**Pull request merged**